scripts/globals/chocobo.lua

```diff
@@ -118,8 +150,21 @@ xi.chocobo.renterOnEventFinish = function(player, csid, option, eventSucceed)
         local mLvl   = player:getMainLvl()
         local zoneId = player:getZoneID()
         local info   = chocoboInfo[zoneId]
+        local trade  = player:getLocalVar('Chocopass')
+        local duration = 900
 
-        if info then
+        if not info then
+            printf('[warning] player %s passed bad zoneId %i in xi.chocobo.renterOnEventFinish', player:getName(), zoneId)
+            return
+        end
+
+        if trade == 1 then-- If the player used a Chocopass/Chocobo Ticket
+            player:setLocalVar('Chocopass', 0)
+            player:setLocalVar('ChocopassDuration', 0)
+            player:tradeComplete()
```

Comment:
If you can do a trade with more than 1 quantity (e.g. 2 chocobo tickets) and it still succeeds then `trade:hasItemQty` isn't the right approach. If you can have various other items in the trade window and it succeeds then you shouldn't be using `player:tradeComplete()` unless all of those other items are also consumed. You should probably be using `npcUtil.tradeHas` and `player:confirmTrade()`

Reply:
updated to `npcUtil.tradeHasExactly` 🙏🏼